### ISSUE TITLE
Class violates singleton/factory pattern definition for traits

### ISSUE THREAD

**Issue by Fleshgrinder** [kalessil/phpinspectionsea#129]
This inspection should ignore traits completely, which it does not right now.

```php
trait MWE {

    protected function __construct() {}

}
```
